src/components/Invoices/InvoicePaymentForm/formUtils.ts

```diff
@@ -0,0 +1,53 @@
+import { type DedicatedInvoicePaymentForm, type Invoice } from '../../../features/invoices/invoiceSchemas'
+import { BigDecimal as BD } from 'effect'
+import { convertBigDecimalToCents, convertCentsToBigDecimal } from '../../../utils/bigDecimalUtils'
+import { formatDate, startOfToday } from 'date-fns'
+import { getLocalTimeZone, fromDate, toCalendarDate } from '@internationalized/date'
+import { DATE_FORMAT_SHORT } from '../../../config/general'
+
+export const getInvoicePaymentFormDefaultValues = (invoice: Invoice): DedicatedInvoicePaymentForm => {
+  const paidAt = fromDate(startOfToday(), getLocalTimeZone())
+
+  return {
+    amount: convertCentsToBigDecimal(invoice.outstandingBalance),
+    method: null,
+    paidAt,
+    referenceNumber: '',
+    memo: '',
+  }
+}
+
+export const validateInvoicePaymentForm = ({ invoicePayment, invoice }: { invoicePayment: DedicatedInvoicePaymentForm, invoice: Invoice }) => {
+  const { amount, paidAt, method } = invoicePayment
+
+  const errors = []
+  if (!BD.isPositive(amount)) {
+    errors.push({ amount: 'Payment amount must be greater than zero.' })
+  }
+
+  if (BD.greaterThan(amount, convertCentsToBigDecimal(invoice.outstandingBalance))) {
+    errors.push({ amount: 'Payment amount cannot be greater than the outstanding invoice balance.' })
+  }
+
+  if (paidAt === null) {
+    errors.push({ paidAt: 'Payment date is a required field.' })
+  }
+
+  if (invoice.sentAt && toCalendarDate(paidAt).compare(toCalendarDate(fromDate(invoice.sentAt, 'UTC'))) < 0) {
+    errors.push({ paidAt: `Payment date cannot be before the invoice date (${formatDate(invoice.sentAt, DATE_FORMAT_SHORT)}).` })
+  }
+
+  if (method === null) {
+    errors.push({ method: 'Payment method is a required field.' })
+  }
```

Comment:
Not for today: this feels like a place where a validation library can make this easier, i.e. a zod. Is ther ea reason we're not using one here with our tanstack forms?